File: app/src/main/java/ru/samtakoy/core/di/modules/AppModule.java
package ru.samtakoy.core.di.modules;

import android.content.ContentResolver;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;


@Module
public class AppModule {

    private Context mContext;

    public AppModule(Context context) {
        this.mContext = context;
    }

    @Provides
    @Singleton
    public Context provideContext() {
        return mContext;
    }

    @Provides
    @Singleton
    public ContentResolver provideContentResolver() {
        return mContext.getContentResolver();
    }

    /*
    @Provides
    @Singleton
    public MyRoomDb getDataBase()  {
        return Room.databaseBuilder(mContext, MyRoomDb.class, DB_NAME).build();
    }*/

}

File: app/src/main/java/ru/samtakoy/core/app/di/modules/AppModule.kt
package ru.samtakoy.core.app.di.modules

import android.content.ContentResolver
import android.content.Context
import dagger.Module
import dagger.Provides
import ru.samtakoy.core.app.some.Resources
import ru.samtakoy.core.app.some.ResourcesImpl
import javax.inject.Singleton

@Module
class AppModule(
    private val mContext: Context
) {
    @Provides @Singleton fun provideContext(): Context {
        return mContext
    }

    @Provides @Singleton fun provideContentResolver(): ContentResolver {
        return mContext.getContentResolver()
    }

    @Provides @Singleton fun provideResources(resources: ResourcesImpl): Resources {
        return resources
    }
}
